### DISCUSSION TITLE
SecurityOnion 2.3.130 soup upgrade to .140 not working

### DISCUSSION THREAD

**Discussion by TotieBash** [Security-Onion-Solutions/securityonion#8399] [General] [answered]
SecurityOnion 2.3.130 soup upgrade to .140 not working with errors below.  I checked /root/soup.log but nothing obvious is sticking out.  The only error I see is "**ERROR: salt-master could not be reached....."**  This is my lab virtual machine so I have reverted snapshot 3 times now and I get to the same error.  I check the md5sum of the iso and compare it with the download page and it looks good.  Help pls....

- SYSTEM INFO:  SecurityOnion 2.3.130 Standalone
- NEW UPDATE ISO FILE:  securityonion-2.3.140-20220719.iso
- SO INSTALL TYPE:  SecurityOnion ISO image install
- SO INSTALL TYPE:  air-gap install

![Copying new Security Onion code from tmp50ag-pdateSecurityOnion to optsosaltstackdefault](https://user-images.githubusercontent.com/86748904/181832284-04ea7192-2c3f-4f68-82e6-57389e9b7aae.png)

**Comment by reyesj2** [comment]
Can you ping your saltmaster
```
sudo salt '*' test.ping
```

**Comment by TotieBash** [comment]
![image](https://user-images.githubusercontent.com/86748904/181808449-164455fc-9003-4190-941d-49254fd7c0a1.png)

**Comment by TotieBash** [comment, marked as answer]
Ok I got pass the first issue. @reyesj2 reply pointed me to the right direction.  The theory is salt was being impatient and not waiting for salt to fully start.   So needed to insert "sleep 120" somewhere in soup.  The caveat is /sbin/soup is guarded by salt so I can't just tweak it so here is the process that I did.

1.  Run the first phase of 'soup -y -f /dev/sr0'.  This process updates the /sbin/soup file to the latest.  '/dev/sr0' is my media drive where the latest iso.
2.  copy the latest /sbin/soup /home/admin
I copied mine to /home/admin/
3.  vim /home/admin/soup and insert "sleep 120" (for mine 120 seconds was enough)
![Pasted Graphic 7](https://user-images.githubusercontent.com/86748904/181839033-a2c427d3-5619-4f32-b7f5-9ee92784d2b3.png)
4. Run the second phase soup using the edited soup file: /home/admin/soup -y -f /dev/sr0
The installer now pass the section where I had troubles with
![Pasted Graphic 8](https://user-images.githubusercontent.com/86748904/181839394-8595e475-d51b-42ed-81eb-079104cc7c28.png)

However I am now on a new error but I think this is the same filebeat error in #8324 and others had issue.  I just need to dig around for the fix.  If someone can link me to the fix I would really appreciate it.
![Pasted Graphic 9](https://user-images.githubusercontent.com/86748904/181839552-b2fc4079-a4a5-4001-ad7b-a09f84df700a.png)

**Comment by dougburks** [comment]
Have you previously modified your Filebeat configuration in `/opt/so/saltstack/local/salt/filebeat/`?

If so, try reverting to snapshot and then removing your Filebeat modifications.

Check this discussion for additional details:
https://github.com/Security-Onion-Solutions/securityonion/discussions/8372

**Reply by TotieBash** [reply]
Yes.  I edited '/opt/so/saltstack/local/salt/filebeat/init.sls' to enable Cisco IOS-XE module in ingesting logs.
![image](https://user-images.githubusercontent.com/86748904/181842178-32fa2721-0b9a-4930-ad5f-1a922be67615.png)

**Reply by TotieBash** [reply]
ok I will try that..

**Reply by TotieBash** [reply]
Ok. It does not look like it succeeded.  Or does it?
![atready](https://user-images.githubusercontent.com/86748904/181852143-e72f1104-7a5a-41bc-b27f-5b717c46f72f.png)

so-status looks good
![image](https://user-images.githubusercontent.com/86748904/181853281-23cf65fc-5113-4cff-a60c-300c33a842de.png)

docker ps looks like they are running the latest .140
![Pasted Graphic 23](https://user-images.githubusercontent.com/86748904/181852691-7c2ea8f0-9837-4b4b-b478-58d44077aea9.png)

docker images looks like updated
![Pasted Graphic 22](https://user-images.githubusercontent.com/86748904/181852437-9fc4e904-f686-4b37-ae8f-aa3ddf2231ed.png)

I can get to SOC and /grafana.  However /kibana is not working.  I will let it marinate for 15 minutes and then I will reboot it.  I will dig through other threads I think I saw someone with this same Kibana issue afeter 140 upgrade.  If someone has idea please send me link.  Thanks @dougburks for directing me to the fix.

**Reply by TotieBash** [reply]
ok /kibana is back to normal again.

This fixed it for me #8335
![weslambert 10 days ago](https://user-images.githubusercontent.com/86748904/181855017-a92bf2ed-fb34-41ff-b19f-3cc4248ca110.png)

My Elastic Auth is already enabled so I did not need this one
![root@zeus-so2  # so-elastic-auth true](https://user-images.githubusercontent.com/86748904/181855078-1d282cf4-c4c0-4146-815b-e97d290f7d91.png)

![syncing users and roles between soc and Elastic](https://user-images.githubusercontent.com/86748904/181855087-ff38b8af-4ed2-4aca-8db7-9250c99d8cfa.png)

Thanks you @weslambert and the entire SecurityOnion team..
